File: documentation/Exercise 5/Java/src/main/java/de/thu/recipebook/models/RecipeCollection.java
package de.thu.recipebook.models;

import java.util.ArrayList;
import java.util.List;

public class RecipeCollection {
    private static RecipeCollection instance;

    private List<Recipe> recipes;

    private RecipeCollection() {
        recipes = new ArrayList<>();
    }

    public static RecipeCollection getInstance() {
        if (instance == null) {
            synchronized (RecipeCollection.class) {
                instance = new RecipeCollection();
            }
        }
        return instance;
    }

    public List<Recipe> getRecipes() {
        return recipes;
    }

    public void setRecipes(List<Recipe> recipes) {
        this.recipes.clear();
        this.recipes.addAll(recipes);
    }
}

File: Exercise 5/Kotlin/src/main/java/de/thu/recipebook/models/RecipeCollection.kt
package de.thu.recipebook.models

class RecipeCollection private constructor() {
    private val recipes: MutableList<Recipe>

    companion object {
        var instance: RecipeCollection? = null
            get() {
                if (field == null) {
                    synchronized(RecipeCollection::class.java) { field = RecipeCollection() }
                }
                return field
            }
            private set
    }

    init {
        recipes = ArrayList()
    }

    fun getRecipes(): MutableList<Recipe> {
        return recipes
    }

    fun setRecipes(recipes: MutableList<Recipe>) {
        this.recipes.clear()
        this.recipes.addAll(recipes)
    }
}